flake.nix

```diff
@@ -24,372 +23,81 @@
-                hash = "sha256-+k9Em+xX1rWPs3oATy3g1U0O6y3CATT9P42p0YCafxM=";
-                leaveDotGit = false;
-              })
-            ];
-
-            setupHook = ./prjxray-setup-hook.sh;
-
-            nativeBuildInputs = [ cmake git ];
-            buildInputs = [ python310Packages.boost python310 eigen ];
-
-            installPhase = ''
-              mkdir -p $out/bin
-              cp -v tools/xc7frames2bit tools/xc7patch $out/bin
-              mkdir -p $out/usr/share/python3/
-              cp -rv $srcs/prjxray $out/usr/share/python3/
-            '';
-
-            doCheck = false;
-
-            meta = with lib; {
-              description = "Xilinx series 7 FPGA bitstream documentation";
-              homepage = "https://github.com/f4pga/prjxray";
-              license = licenses.isc;
-              platforms = platforms.all;
-              maintainers = with maintainers; [ thoughtpolice ];
+            kintex7 = callPackage ./nix/nextpnr-xilinx-chipdb.nix {
+              backend = "kintex7";
+            };
+            spartan7 = callPackage ./nix/nextpnr-xilinx-chipdb.nix {
+              backend = "spartan7";
+            };
+            zynq7 = callPackage ./nix/nextpnr-xilinx-chipdb.nix {
+              backend = "zynq7";
             };
           };
+        });
 
-        nextpnr-xilinx-chipdb = {
-          artiz7 = prev.callPackage ./nix/nextpnr-xilinx-chipdb.nix {
-            backend = "artiz7";
-          };
-          kintex7 = prev.callPackage ./nix/nextpnr-xilinx-chipdb.nix {
-            backend = "kintex7";
-          };
-          spartan7 = prev.callPackage ./nix/nextpnr-xilinx-chipdb.nix {
-            backend = "spartan7";
-          };
-          zynq7 = prev.callPackage ./nix/nextpnr-xilinx-chipdb.nix {
-            backend = "zynq7";
-          };
-        };
-      };
-
-      # Provide some binary packages for selected system types.
-      packages = forAllSystems (system: {
-        inherit (nixpkgsFor.${system})
-          yosys ghdl yosys-ghdl prjxray nextpnr-xilinx nextpnr-xilinx-chipdb;
-      });
-
-      # The default package for 'nix build'. This makes sense if the
-      # flake provides only one package or there is a clear "main"
-      # package.
-      defaultPackage = forAllSystems (system: self.packages.${system}.yosys);
-
+      # FIXME(jl): what is the actual intended workflow with this dev shell?
+      # what's the intended use case of this, should it contain all packages for a pipeline?
+      # are these all mutually dependent on one another, or does it make sense to use any/some/all of these
+      # independently
+      #
+      # for example, we can build and run `nextpnr-xilinx` on its own independent of the override yosys version.
+      # is this at all useful, or is this tool dependent on this version of yosys, and otherwise useless?
+      # in which case, consider making it a hard runtime dependency or something that is enabled by default
```

Comment:
calling out this comment and question as well

Reply:
This is how the toolcain is used (with my current snap package)
https://github.com/openXC7/demo-projects/blob/main/blinky-qmtech/Makefile

Reply:
This tool currently depends on yosys 0.17 as input, because later yosys versions generate BRAM primitives that nextpnr-xilinx does not yet understand. Once those are implemented, we can use the latest yosys versions.

Reply:
> Once those are implemented, we can use the latest yosys versions.

Do you have a sense of when those BRAM primitives would be implemented in `nextpnr-xilinx`?

Upstreaming older versions of packages can be somewhat complicated and difficult. As an example, [OpenJDK has a derivation defined for each version](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/compilers/openjdk), so it is possible, but more likely it'd be easier to upstream `nextpnr-xilinx` once it can rely on a newer version of yosys.